csrc/scheduler/hopper_multi_matmul.cpp

```diff
@@ -643,41 +641,24 @@ void HopperMultipleMatmulScheduler::scheduleSplitKSum() {
   if (params_->splitk_factor == 1) {
     return;
   }
-  NVF_THROW("Split-K scheduling is not yet implemented for Hopper matmul");
   for (TensorView* splitk_sum : splitk_sums_) {
     // Always use serial grid reduction for split-K sum
     splitk_sum->definition()->as<ReductionOp>()->requestSerialGridReduction();
 
-    if (params_->use_smem_epilogue) {
-      // Now that transforms are propagated backward to smem_epilogue, which
-      // is before splitk_sum, we can vectorize the inner-most non-trivial
-      // dimension of splitk_sum
-      //
-      // Note that the split-K reduction is the inner-most dimension.
-      Val* vec_ext = splitk_sum->axis(-2)->extent();
-      NVF_ERROR(vec_ext->isConstInt());
-      int64_t vec_ext_int = vec_ext->evaluate().as<int64_t>();
-      splitk_sum->axis(-1)->parallelize(ParallelType::BIDz);
-      splitk_sum->axis(-3)->parallelize(ParallelType::TIDx);
-      if (vec_ext_int * dataTypeSize(splitk_sum->dtype()) > 16) {
-        // NOTE: We might encounter an illegal vectorization size if we are
-        // using Float for this reduction and Half for output. So here we
-        // first check whether the vectorize size is at most 16 bytes. If not,
-        // then we split into an unrolled loop that will do multiple
-        // vectorized reads/writes instead. Note that we reorder such that the
-        // axes are in order UR TIDx V.
-        splitk_sum->split(
-            -2, 16 / dataTypeSize(splitk_sum->dtype()), /*inner_split=*/true);
-        splitk_sum->axis(-3)->parallelize(ParallelType::Unroll);
-        splitk_sum->reorder({{-4, -3}});
-        // In this case, we have [... iUR iTx rBz iS]
-      }
-      splitk_sum->reorder({{-2, -1}});
-    } else { // no smem epilogue
-      // Reorder to place the split-K reduction next to innermost [... rBz iS]
-      splitk_sum->reorder({{-9, -2}});
-    }
-    // Vectorize inner-most dimension [... (iUR iTx) rBz iV]
+    // [..., Mo, No, Mi, Ni]
+    splitk_sum->split(-2, getM(params_->mma_macro));
+    splitk_sum->split(-1, getN(params_->mma_macro));
+    // [..., Mo, No, Mio, Mii, Nio, Nii]
+    // -> [..., Mo, No, Mio, Nio, Mii, Nii]
+    splitk_sum->reorder({{-3, -2}});
+    splitk_sum->merge(-4);
+    auto s = mma_utils::MmaSwizzler::scheduleMmaOutputAllocation(
+        splitk_sum->getLoopDomain());
+    splitk_sum->setLoopDomain(s.as<IterDomain*>());
+    splitk_sum->axis(-5)->parallelize(ParallelType::TIDy);
+
+    // splitk_sum->reorder({{2, -2}});
+    splitk_sum->axis(2)->parallelize(ParallelType::BIDz);
```

Comment:
Not for this PR: this is just a note for future work. Below here you can see that this will use a vectorization width of 2, whereas for split-K we should always have tile sizes whose N is a multiple of 8, so we can always achieve full vectorization by unswizzling in smem first like we do on Ampere. I think we should probably always do that when `params.use_smem_epilogue` is true since it will not increase smem use in that case so there's no reason not to. Anyway, as I say that should definitely be kept for a later PR.